src/cascadia/UnitTests_SettingsModel/SerializationTests.cpp

```diff
@@ -948,4 +952,127 @@ namespace SettingsModelUnitTests
 
         VERIFY_ARE_EQUAL(toString(newResult), toString(oldResult));
     }
+
+    void SerializationTests::RoundtripGenerateActionID()
+    {
+        static constexpr std::string_view oldSettingsJson{ R"(
+        {
+            "actions": [
+                {
+                    "name": "foo",
+                    "command": { "action": "sendInput", "input": "just some input" },
+                    "keys": "ctrl+shift+w"
+                }
+            ]
+        })" };
+
+        // Key differences: - the sendInput action now has a generated ID
+        //                  - this generated ID was created at the time of writing this test,
+        //                    and should remain robust (i.e. every time we hash the args we should get the same result)
+        static constexpr std::string_view newSettingsJson{ R"(
+        {
+            "actions": [
+                {
+                    "name": "foo",
+                    "command": { "action": "sendInput", "input": "just some input" },
+                    "keys": "ctrl+shift+w",
+                    "id" : "User.sendInput.3448838294654165202"
+                }
+            ]
+        })" };
+
+        implementation::SettingsLoader oldLoader{ oldSettingsJson, implementation::LoadStringResource(IDR_DEFAULTS) };
+        oldLoader.MergeInboxIntoUserSettings();
+        oldLoader.FinalizeLayering();
+        VERIFY_IS_TRUE(oldLoader.FixupUserSettings(), L"Validate that this will indicate we need to write them back to disk");
+        const auto oldSettings = winrt::make_self<implementation::CascadiaSettings>(std::move(oldLoader));
+        const auto oldResult{ oldSettings->ToJson() };
+
+        implementation::SettingsLoader newLoader{ newSettingsJson, implementation::LoadStringResource(IDR_DEFAULTS) };
+        newLoader.MergeInboxIntoUserSettings();
+        newLoader.FinalizeLayering();
+        newLoader.FixupUserSettings();
+        const auto newSettings = winrt::make_self<implementation::CascadiaSettings>(std::move(newLoader));
+        const auto newResult{ newSettings->ToJson() };
+
+        VERIFY_ARE_EQUAL(toString(newResult), toString(oldResult));
+    }
+
+    void SerializationTests::DoNotGenerateActionID()

```

Comment:
nit: test names - _why_ are we doing this? We don't just want to test "don't generate action ID", we want to test "Action IDs for Iterated Commands" or something